kedro_azureml/hooks.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+
+from kedro.framework.hooks import hook_impl
+
+from kedro_azureml.datasets.folder_dataset import AzureMLFolderDataSet
+from kedro_azureml.runner import AzurePipelinesRunner
+
+
+class AzureMLLocalRunHook:
+    """Hook class that allows local runs using AML datasets."""
+
+    @hook_impl
+    def before_pipeline_run(self, run_params, pipeline, catalog):
+        """Hook implementation to change dataset path for local runs.
+        Args:
+            run_params: The parameters that are passed to the run command.
+            pipeline: The ``Pipeline`` object representing the pipeline to be run.
+            catalog: The ``DataCatalog`` from which to fetch data.
+        """
+        # we don't want the hook to work when we are running on AML
+        if not isinstance(run_params["runner"], AzurePipelinesRunner):
```

Comment:
It seems `run_params["runner"]` is passed the name of the runner, so I would change this to
```suggestion
        if run_params["runner"] != "AzurePipelinesRunner":
```